test_dgemmt (actually DGEMM) failure on ppc64le with GCC 15

Looks like the `test_extensions/test_dgemmt` test has some failure when OpenBLAS is built with GCC 15 on ppc64le (Fedora 42). Building with GCC 14 (Fedora 41) there is no such issue.
````
...
TEST 1197/1522 dgemmt:c_api_rowmajor_lower_beta_one [OK]
TEST 1198/1522 dgemmt:c_api_rowmajor_lower_alpha_zero [OK]
TEST 1199/1522 dgemmt:c_api_rowmajor_lower_M_50_K_50_a_trans_b_trans [OK]
TEST 1200/1522 dgemmt:c_api_rowmajor_lower_M_50_K_100_a_notrans_b_trans [OK]
TEST 1201/1522 dgemmt:c_api_rowmajor_lower_M_100_K_50_a_trans_b_notrans [OK]
TEST 1202/1522 dgemmt:c_api_rowmajor_lower_M_50_K_50_a_notrans_b_notrans [OK]
TEST 1203/1522 dgemmt:c_api_rowmajor_upper_beta_one [OK]
TEST 1204/1522 dgemmt:c_api_rowmajor_upper_alpha_zero [OK]
TEST 1205/1522 dgemmt:c_api_rowmajor_upper_M_50_K_50_a_trans_b_trans [OK]
TEST 1206/1522 dgemmt:c_api_rowmajor_upper_M_50_K_100_a_notrans_b_trans [OK]
TEST 1207/1522 dgemmt:c_api_rowmajor_upper_M_100_K_50_a_trans_b_notrans [FAIL]
  ERR: test_extensions/test_dgemmt.c:820  expected 0.000e+00, got 6.352e-02 (diff -6.352e-02, tol 1.000e-13)
TEST 1208/1522 dgemmt:c_api_rowmajor_upper_M_50_K_50_a_notrans_b_notrans [OK]
TEST 1209/1522 dgemmt:c_api_colmajor_lower_beta_one [OK]
TEST 1210/1522 dgemmt:c_api_colmajor_lower_alpha_zero [OK]
TEST 1211/1522 dgemmt:c_api_colmajor_lower_M_50_K_50_a_trans_b_trans [OK]
TEST 1212/1522 dgemmt:c_api_colmajor_lower_M_50_K_100_a_notrans_b_trans [OK]
TEST 1213/1522 dgemmt:c_api_colmajor_lower_M_100_K_50_a_trans_b_notrans [OK]
TEST 1214/1522 dgemmt:c_api_colmajor_lower_M_50_K_50_a_notrans_b_notrans [OK]
TEST 1215/1522 dgemmt:c_api_colmajor_upper_beta_one [OK]
TEST 1216/1522 dgemmt:c_api_colmajor_upper_alpha_zero [OK]
TEST 1217/1522 dgemmt:c_api_colmajor_upper_M_50_K_50_a_trans_b_trans [OK]
TEST 1218/1522 dgemmt:c_api_colmajor_upper_M_50_K_100_a_notrans_b_trans [OK]
TEST 1219/1522 dgemmt:c_api_colmajor_upper_M_100_K_50_a_trans_b_notrans [OK]
TEST 1220/1522 dgemmt:c_api_colmajor_upper_M_50_K_50_a_notrans_b_notrans [OK]
TEST 1221/1522 dgemmt:lower_beta_one [OK]
TEST 1222/1522 dgemmt:lower_alpha_zero [OK]
TEST 1223/1522 dgemmt:lower_M_50_K_50_a_trans_b_trans [OK]
TEST 1224/1522 dgemmt:lower_M_50_K_100_a_notrans_b_trans [FAIL]
  ERR: test_extensions/test_dgemmt.c:431  expected 0.000e+00, got 2.585e-01 (diff -2.585e-01, tol 1.000e-13)
TEST 1225/1522 dgemmt:lower_M_100_K_50_a_trans_b_notrans [OK]
TEST 1226/1522 dgemmt:lower_M_50_K_50_a_notrans_b_notrans [OK]
TEST 1227/1522 dgemmt:upper_beta_one [OK]
TEST 1228/1522 dgemmt:upper_alpha_zero [OK]
...
RESULTS: 1522 tests (1520 ok, 2 failed, 0 skipped) ran in 973 ms
make[1]: *** [Makefile:87: run_test] Chyba 2
make[1]: Opouští se adresář „/mnt/dan/OpenBLAS/utest“
make: *** [Makefile:177: tests] Chyba 2
````
In  both cases I am doing `make TARGET=POWER9 `, but it fails with `POWER8` kernel as well.

--- a/kernel/power/dgemv_n.c
+++ b/kernel/power/dgemv_n.c
@@ -147,7 +147,7 @@ int CNAME(BLASLONG m, BLASLONG n, BLASLONG dummy1, FLOAT alpha, FLOAT *a, BLASLO
 	BLASLONG m3;
 	BLASLONG n2;
 	BLASLONG lda4 =  lda << 2;
-	FLOAT xbuffer[8] __attribute__ ((aligned (16)));
+	volatile FLOAT xbuffer[8] __attribute__ ((aligned (16)));
 	FLOAT *ybuffer;
 
         if ( m < 1 ) return(0);

